neo/Consensus/ConsensusMessage.cs

```diff
@@ -24,7 +24,11 @@ protected ConsensusMessage(ConsensusMessageType type)
 
         public virtual void Deserialize(BinaryReader reader)
         {
-            if (Type != (ConsensusMessageType)reader.ReadByte())
+            ConsensusMessageType cmt = (ConsensusMessageType)reader.ReadByte();
+            if (!Enum.IsDefined(typeof(ConsensusMessageType), cmt))
```

Comment:
Move this lines to the constructor

Reply:
I don't understand the suggestion. The type needs to be read from the BinaryReader and the result (`cmt`) has 2 usages
1.  check for an invalid enum type
2. next check if it equals to whatever `Type` the message already had.

Reply:
If you check that the Type is a valid one in the constructor, you don't need to change this code.

Reply:
@ixje In this particular case it will check that the value is exactly the same as the constructor. So i think that is not needed.